db/migrate/20220520105024_update_service_user_roles.rb

```diff
@@ -0,0 +1,38 @@
+class UpdateServiceUserRoles < ActiveRecord::Migration[6.0]
+  class MiqUserRole < ActiveRecord::Base
+    has_and_belongs_to_many :miq_product_features, :join_table => :miq_roles_features, :class_name => "UpdateServiceUserRoles::MiqProductFeature"
+  end
+
+  class MiqProductFeature < ActiveRecord::Base; end
+
+  def up
+    say_with_time("Correcting user created role feature sets") do
+      service         = MiqProductFeature.find_by(:identifier => "service")
+      service_accord  = MiqProductFeature.find_by(:identifier => "service_accord")
+
+      if service_accord
+        affected_user_roles(service_accord).each do |user_role|
+          if user_role.miq_product_features.include?(service_accord) # useless?
```

Comment:
lets see if this all works with this line removed.
I am assuming that the affected_user_roles will filter so this should not be necessary